src/pages/permissions/panels/PermissionSelector.tsx

```diff
@@ -62,39 +72,66 @@ const PermissionSelector: FC<Props> = ({ onAddPermission }) => {
 
   const isLoading = isPermissionsLoading || isMetadataLoading;
 
+  // focus permission selector container on first render so tabbing will consistently navigate to resource type selector first
   useEffect(() => {
-    document.getElementById("resourceType")?.focus();
+    setTimeout(() => {
+      if (permissionSelectorRef.current) {
+        permissionSelectorRef.current.focus();
+        permissionSelectorRef.current.tabIndex = -1;
+      }
+    }, 100);
   }, []);
 
-  useEffect(() => {
-    if (!resourceType) {
-      document.getElementById("resourceType")?.focus();
-      return;
+  const focusAfterResourceTypeChange = (
+    prevResourceType: string,
+    currResourceType: string,
+    permissions?: LxdPermission[],
+  ) => {
+    // for server resource type, there will be no resources to select
+    // skip to entitlement selector
+    if (currResourceType === "server") {
+      entitlementRef.current?.open();
+      return currResourceType;
     }
 
-    if (resourceType === "server") {
-      document.getElementById("entitlement")?.focus();
-      return;
+    // in case if the same resource type is selected
+    if (prevResourceType && currResourceType === prevResourceType) {
+      // open the resource dropdown if there are permissions
+      if (permissions?.length) {
+        resourceRef.current?.open();
+      } else {
+        // keep focus on the resource type selector if there are no permissions
+        resourceTypeRef.current?.focus();
+      }
+      return currResourceType;
     }
-    document.getElementById("resource")?.focus();
-  }, [resourceType, permissions]);
 
-  useEffect(() => {
-    if (resource) {
-      document.getElementById("entitlement")?.focus();
+    // In case if the resource type is changed
+    if (!prevResourceType && currResourceType) {
+      // open the resource dropdown if there are permissions for the resource type
+      if (permissions?.length) {
+        resourceRef.current?.open();
+        return currResourceType;
+      }
+
+      // if there are no permissions, we can't select a resource
+      // keep focus on the resource type selector
+      resourceTypeRef.current?.focus();
```

Comment:
```suggestion
  const focusAfterResourceTypeChange = (
    previous: string,
    current: string,
    permissions?: LxdPermission[],
  ) => {
    // for server, there will be no resources to select
    // skip to entitlement selector
    if (current === "server") {
      entitlementRef.current?.open();
      return current;
    }

    const wasChanged = !previous && current;
    const wasReselected = previous && current === previous;
    if (wasChanged || wasReselected) {
      // open the resource dropdown if there are permissions for the resource type
      if (permissions?.length) {
        resourceRef.current?.open();
        return current;
      }

      // if there are no permissions, we can't select a resource
      // keep focus on the resource type selector
      resourceTypeRef.current?.focus();
    }

    return current;
  };
```
We can remove the duplication and should aim for self documenting code in favour of too many comments.